docs/articles/documentation/test-api/obtaining-data-from-the-client/examples-of-using-client-functions.md

```diff
@@ -0,0 +1,173 @@
+        //Test logic for Safari
+        console.log('The browser is Safari');
+    }
+});
+```
+
+## Injecting External Libraries into a Page from a Test
+
+One more case of using client functions is injecting external libraries like [jQuery](https://jquery.com/) into a tested page.
+
+In order to work with jQuery from a test, you first need to add jQuery code to the tested page by using the `eval` method.
+
+```js
+await t.eval(new Function(fs.readFileSync('./jquery.js').toString()));
+```
+
+After that,  you can use jQuery in the test within a client function.
+
+```js
+import { ClientFunction } from 'testcafe';
+import fs from 'fs';
+
+fixture `My fixture`
+    .page `http://devexpress.github.io/testcafe/example/`;
+
+test('My test', async t => {
+    // Adds jQuery code to the page
+    await t.eval(new Function(fs.readFileSync('./jquery.min.js').toString()));
+
+    const getElementText = ClientFunction(() => {
+        // Uses the jQuery selector
+        return $('div.column.col-2').find('legend').eq(0).text();
+    });
+
+    await t.expect(getElementText()).contains('What is your primary Operating System');
+});
+```
+
+## Accessing Child Nodes in the DOM Hierarchy
+
+To find all child elements (not [nodes](https://developer.mozilla.org/en-US/docs/Web/API/Node)) of nodes in a DOM hierarchy, TestCafe provides the [Selector.child](../selecting-page-elements/selectors/functional-style-selectors.md#child) method. However, you may need to access child nodes of some node. You can do this by using a client function.
+
+Suppose that you test the *example.html* page with the following HTML code.
+
+```js
+<!DOCTYPE html>
+<html>
+    <body id="testedpage">
+        This is my tested page. <!--This is the first child node of <body>-->
+        <p>My first paragraph.</p>
+        <p>My second paragraph.</p>
+    </body>
+</html>
+```
+
+Also suppose that you need to obtain the text *This is my tested page*. This text is considered as a child node, not a child element of the body element. So, you can’t access this text by using `Selector.child`, but can obtain the [childNodes](https://developer.mozilla.org/en-US/docs/Web/API/Node/childNodes) property from the client side.
+
+```js
+import { Selector } from 'testcafe';
+import { ClientFunction } from 'testcafe';
+
+fixture `My Fixture`
+    .page `examplepage.html`;
+
+    const getChildNodeText = ClientFunction(index => {
+        return getEl().childNodes[index].textContent;
+    },{ dependencies: { getEl: Selector('#testedpage') } });
+
+
+test('My Test', async t => {
+    console.log(await getChildNodeText(0));
+});
+```
+
+## Complex DOM Queries
+
+Client functions can be helpful for complex DOM queries. For example, a tested page contains a table with some data and you need to validate data from particular two columns only. To do this, you can obtain data from these columns, push them to an array and then compare the returned array with the expected one.
```

Comment:
I still think that `two particular` is grammatically better. Again, talk with the correctors please.